### PR TITLE
Added a new class to handle non fatal errors.

### PR DESCRIPTION
**Pull request by PaulaRudy**
This is associated with trello card 86: https://trello.com/c/GbrSIOPr

### CODE REVIEW

File: src/main/java/edu/wpi/grip/core/events/NotificationEvent.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class NotificationEvent {
+    private final Throwable exception;
+    private final Optional<String> detailedDescription;
+
+    public NotificationEvent(Throwable exception){
+        this(exception, null);
+    }
+
+    public NotificationEvent(Throwable exception, String detailedDescription) {
+        this.exception = exception;
```
Comment (JLLeitschuh) [inline src/main/java/edu/wpi/grip/core/events/NotificationEvent.java]: Use `checkNotNull(exception, "The exception can not be null");` from googles api

**Comment by JLLeitschuh** [comment]
Also, look at the FXML changes I added to the main window:
Can you implement those too: https://github.com/WPIRoboticsProjects/GRIP/pull/84/files
Also, the event handing in the main window.

**Comment by PaulaRudy** [comment]
Shouldn't you do this then? Since it's really just re-implementing something you already did?

**Comment by PaulaRudy** [comment]
I mean, if I'm really just copy-pasting this, shouldn't you just be getting the old pull request up to date and we can merge that instead?

**Comment by JLLeitschuh** [comment]
That PR is very very out of date. I don't have the time to bring it up to date. (Thats why I asked you to do it). If I was redoing it I would be copying and pasting and creating a new PR anyways.
You just need to implement the event handler inside of the controller.

**Comment by PaulaRudy** [comment]
Ahhh Ok

**Pull request closed without merging**